crates/anvil/src/eth/backend/mem/mod.rs

```diff
@@ -1797,10 +1797,10 @@ impl Backend {
             }
         }
         let final_block_request = match block_request {
-            Some(BlockRequest::Pending(_)) => Some(BlockRequest::Number(self.best_number())),
+            Some(BlockRequest::Pending(_)) | None => Some(BlockRequest::Number(self.best_number())),
             Some(BlockRequest::Number(bn)) => Some(BlockRequest::Number(bn)),
-            None => None,
         };
```

Comment:
this is always Some now?

Reply:
ye, would just do the `Some` in the call to `with_database_at` as a nit

Reply:
actually, we don't need the option here at all